board/stm32fx/llflash.h

```diff
@@ -0,0 +1,27 @@
+bool flash_unlock(void) {
+  if (FLASH->CR & FLASH_CR_LOCK) {
+    FLASH->KEYR = 0x45670123;
+    FLASH->KEYR = 0xCDEF89AB;
+    return true;
+  }
+  return false;
+}
+
+bool flash_erase_sector(uint8_t sector, bool unlocked) {
+  // don't erase the bootloader(sector 0)
+  if (sector != 0 && sector < 12 && unlocked) {
+    FLASH->CR = (sector << 3) | FLASH_CR_SER;
+    FLASH->CR |= FLASH_CR_STRT;
+    while (FLASH->SR & FLASH_SR_BSY);
+    return true;
+  }
+  return false;
+}
+
+void flash_write_word(void *prog_ptr, const void *data) {
```

Comment:
why pass in the data as a pointer? if you're always writing 32 bits, let's just pass the data in as a uint32_t directly